src/convert.cpp

```diff
@@ -545,7 +545,7 @@ namespace Exiv2 {
         Exiv2::ExifData::iterator pos = exifData_->findKey(ExifKey(from));
         if (pos == exifData_->end()) return;
         if (!prepareXmpTarget(to)) return;
-        for (int i = 0; i < pos->count(); ++i) {
+        for (long i = 0; i < pos->count(); ++i) {
```

Comment:
The return type [`count`](https://github.com/Exiv2/exiv2/blob/fec1813a3e7b0e4aaed7fae7ba9f9b49ca526638/src/exif.cpp#L384) is `long` and the parameter type of [`toString`](https://github.com/Exiv2/exiv2/blob/fec1813a3e7b0e4aaed7fae7ba9f9b49ca526638/src/exif.cpp#L394) is `long`, so `long` is a better choice for the type of `i`.